### PR TITLE
K8SPSMDB-336: Take mongos status to account

### PR DESCRIPTION
**Pull request by blez**
[K8SPSMDB-336](https://jira.percona.com/browse/K8SPSMDB-336)

### CODE REVIEW

File: pkg/controller/perconaservermongodb/status.go [hunk 1]
```diff
@@ -169,6 +169,10 @@ func (r *ReconcilePerconaServerMongoDB) updateStatus(cr *api.PerconaServerMongoD
 		cr.Status.State = api.AppStateError
 	}
 
+	if cr.Status.State != api.AppStateError && cr.Status.Mongos.Status != api.AppStateReady {
```
Comment (delgod) [inline pkg/controller/perconaservermongodb/status.go:172]: cr.Status.Mongos can be nil
```suggestion
	if cr.Status.State != api.AppStateError && cr.Status.Mongos != nil && cr.Status.Mongos.Status != api.AppStateReady {
```

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/controller/perconaservermongodb/status.go (hunk 1 above)
Comment (blez) [reply pkg/controller/perconaservermongodb/status.go]: Thanks!
https://github.com/percona/percona-server-mongodb-operator/pull/494/commits/77f8656f517c95d9973c69e83ffd4d5d9663d744

**Review by nonemax — APPROVED** [review]

**Review by tplavcic — APPROVED** [review]

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-494-77f8656f

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| operator-self-healing | passed |
| security-context | passed |
| scheduled-backup | passed |
| service-per-pod | passed |
| liveness | passed |
| demand-backup-sharded | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| users | passed |
| upgrade-sharded | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| one-pod | failed |
| storage | failed |
| upgrade-consistency | failed |
| demand-backup | failed |
| monitoring-2-0 | passed |
| self-healing | failed |
| limits | failed |
| arbiter | passed |
| scheduled-backup | failed |
| scaling | failed |
| service-per-pod | failed |
| liveness | failed |
| security-context | failed |
| operator-self-healing | failed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | passed |
| smart-update | failed |
| version-service | failed |
| users | failed |

**Pull request merged**